js/src/tooltip.js

```diff
@@ -284,19 +311,37 @@ const Tooltip = (() => {
 
         const container = this.config.container === false ? document.body : $(this.config.container)
 
-        $(tip).data(this.constructor.DATA_KEY, this)
+        $tip.data(this.constructor.DATA_KEY, this)
 
         if (!$.contains(this.element.ownerDocument.documentElement, this.tip)) {
-          $(tip).appendTo(container)
+          $tip.appendTo(container)
         }
 
         $(this.element).trigger(this.constructor.Event.INSERTED)
 
+        let newOffset = this.config.offset
+
+        const arrowOffset = $arrow.width() + this.config.arrowPadding
+
+        if (!newOffset) {
+          switch (OffsetMap[placement.toUpperCase()]) {
+            case +1:
+              newOffset = `+50%p - ${arrowOffset}px`
+              break
+            case -1:
+              newOffset = `-50%p + ${arrowOffset}px`
+              break
+            default:
+              newOffset = 0
+          }
+
+        }
+
         this._popper = new Popper(this.element, tip, {
           placement: attachment,
           modifiers: {
             offset: {
-              offset: this.config.offset
+              offset: newOffset
```

Comment:
by doing that you remove a custom configuration from our documentation see : https://getbootstrap.com/docs/4.0/components/tooltips/#options

Reply:
No - I only sets newOffset if it isn't already set (line 326)